samples/dnn/ssd_mobilenet_object_detection.cpp

```diff
@@ -122,6 +122,27 @@ int main(int argc, char** argv)
                      (fourcc != 0 ? fourcc : VideoWriter::fourcc('M','J','P','G')),
                      (fps != 0 ? fps : 10.0), cropSize, true);
 
+    if (parser.get<bool>("opencl"))
+    {
+        Mat frame;
+        cap >> frame; // get a new frame from camera/video or read image
+        if (frame.channels() == 4)
+            cvtColor(frame, frame, COLOR_BGRA2BGR);
+
+        //! [Prepare blob]
+        Mat inputBlob = blobFromImage(frame, inScaleFactor,
+                                      Size(inWidth, inHeight), meanVal, false); //Convert Mat to batch of images
+        //! [Prepare blob]
+
+        //! [Set input blob]
+        net.setInput(inputBlob, "data"); //set the network input
+        //! [Set input blob]
+
+        //! [Make forward pass]
+        Mat detection = net.forward("detection_out"); //compute output
+        //! [Make forward pass]
```

Comment:
Not sure why we need this, but this can be simplified and placed into the main loop:
```.diff
-for(;;)
+for(size_t N = 0; ; N++)
 {
    ...
     Mat detection = net.forward("detection_out"); //compute output
     //! [Make forward pass]

+    if (N == 0) continue; // warmup on first frame
     ...
 }
...
```

BTW, Comments like these "//! [Set input blob]" are special and should be unique in file

Reply:
yes，my purpose is to ignore first forward and not count its inference time，this seems not necessary in this sample, forward time is count frame by frame, so I just reverted this change